src/TLuaInterpreter.cpp

```diff
@@ -18210,3 +18211,79 @@ int TLuaInterpreter::showNotification(lua_State* L)
     mudlet::self()->mTrayIcon.showMessage(title, text, mudlet::self()->mTrayIcon.icon(), notificationExpirationTime);
     return 0;
 }
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#setMapInfoOverrideCallback
+int TLuaInterpreter::setMapInfoOverrideCallback(lua_State* L) {
+
+    if (lua_gettop(L) > 0 && !lua_isfunction(L, 1) && !lua_isnil(L, 1)) {
+        lua_pushfstring(L, "setMapInfoOverrideCallback: bad argument #1 type (callback as nil or as function expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    }
+
+    bool registeredCallback = false;
+    auto& host = getHostFromLua(L);
+
+    int callback;
+    if(lua_isfunction(L, 1)) {
+        callback = luaL_ref(L, LUA_REGISTRYINDEX);
+    } else {
+        callback = 0;
+    }
+
+    if (host.mpMap) {
```

Comment:
All these can be combined to avoid nesting:
```diff
-     if (host.mpMap) {
-        if (host.mpMap->mpMapper) {
-            if (host.mpMap->mpMapper->mp2dMap) {
-                host.mpMap->mpMapper->mp2dMap->mapInfoOverrideCallback = callback;
-                host.mpMap->mpMapper->mp2dMap->repaint();
-                registeredCallback = true;
-            }
-        }
+     if (host.mpMap && host.mpMap->mpMapper && host.mpMap->mpMapper->mp2dMap) {
+         host.mpMap->mpMapper->mp2dMap->mapInfoOverrideCallback = callback;
+         host.mpMap->mpMapper->mp2dMap->mapInfoOverrideCallback = callback;
+         host.mpMap->mpMapper->mp2dMap->repaint();
+         registeredCallback = true;
      }
```